### PR TITLE
Return PubK in correct type

### PR DESCRIPTION
**Pull request by lukehinds**
PubK requires MarshalPKIXPublicKey and returning as []uint8

Signed-off-by: Luke Hinds <lhinds@redhat.com>

### CODE REVIEW

File: pkg/keymgmt/keymgmt.go [hunk 1]
```diff
@@ -43,31 +44,36 @@ func GeneratePrivateKey(algorithm string) (interface{}, interface{}, error) {
...
 		pub = &k.PublicKey
 	case *ecdsa.PrivateKey:
 		pub = &k.PublicKey
-	// The below would never happen (unless upstream crypto is broken, but lets log it anyway)
 	default:
 		err = errors.New("error generating public key" )
 	}
+	if err != nil {
+		return nil, err
+	}
```
Comment (dlorenc) [inline pkg/keymgmt/keymgmt.go:72]: in some of the cosign code I actually gave up with this error and threw a panic (same as below during MarshalPKIXPublicKey). Like you said, an examination of the code shows no error can be thrown here. It really simplifies callers to not have to deal with this error. A panic seems appropriate. Your call though!

Comment (lukehinds) [reply pkg/keymgmt/keymgmt.go:72]: I tend to agree with that, if the underlying code breaks there is really no recovering and its appropriate to panic.

**Comment by dlorenc** [comment]
Lgtm!

**Comment by lukehinds** [comment]
Thanks @dlorenc , I forget to mention, I dropped RSA for now, as we don't handle it server side

https://github.com/sigstore/fulcio/blob/dbc10540245a5484c5f65419bb61c83b9e34aa22/pkg/ca/ca.go#L50-L60

**Pull request merged**